firedrake/slate/slac/compiler.py

```diff
@@ -0,0 +1,324 @@
+                                                                tensor,
+                                                                coordsym,
+                                                                *clist)],
+                                                   open_scope=True)]))
+                loop = ast.For(ast.Decl("unsigned int", itsym, init=0), ast.Less(itsym, nfacet),
+                               ast.Incr(itsym, 1), loop_body)
+                statements.append(loop)
+            else:
+                statements.append(ast.FunCall(kinfo.kernel.name, tensor, coordsym, *clist))
+
+    # Now we handle any terms that require auxiliary data (if any)
+    if bool(builder.aux_exprs):
+        aux_temps, aux_statements = auxiliary_information(builder)
+        context_temps.update(aux_temps)
+        statements.extend(aux_statements)
+
+    result_sym = ast.Symbol("T%d" % len(builder.temps))
+    result_data_sym = ast.Symbol("A%d" % len(builder.temps))
+    result_type = "Eigen::Map<%s >" % eigen_matrixbase_type(shape)
+    result = ast.Decl(SCALAR_TYPE, ast.Symbol(result_data_sym, shape))
+    result_statement = ast.FlatBlock("%s %s((%s *)%s);\n" % (result_type,
+                                                             result_sym,
+                                                             SCALAR_TYPE,
+                                                             result_data_sym))
+    statements.append(result_statement)
+
+    cpp_string = ast.FlatBlock(metaphrase_slate_to_cpp(slate_expr, context_temps))
+    statements.append(ast.Assign(result_sym, cpp_string))
+
+    args = [result, ast.Decl("%s **" % SCALAR_TYPE, coordsym)]
+    for c in slate_expr.coefficients():
+        if isinstance(c, Constant):
+            ctype = "%s *" % SCALAR_TYPE
+        else:
+            ctype = "%s **" % SCALAR_TYPE
+        args.append(ast.Decl(ctype, builder.coefficient_map()[c]))
+
+    if builder.needs_cell_facets:
+        args.append(ast.Decl("char *", cellfacetsym))
+
+    macro_kernel_name = "compile_slate"
+    kernel_ast, oriented = builder.construct_ast(name=macro_kernel_name,
+                                                 args=args,
+                                                 statements=ast.Block(statements))
+
+    inc.append(petsc.get_petsc_dir() + '/include/eigen3/')
+    op2kernel = op2.Kernel(kernel_ast, macro_kernel_name, cpp=True, include_dirs=inc,
+                           headers=['#include <Eigen/Dense>', '#define restrict __restrict'])
+
+    assert len(slate_expr.ufl_domains()) == 1
+    kinfo = KernelInfo(kernel=op2kernel,
+                       integral_type="cell",
+                       oriented=oriented,
+                       subdomain_id="otherwise",
+                       domain_number=0,
+                       coefficient_map=range(len(slate_expr.coefficients())),
+                       needs_cell_facets=builder.needs_cell_facets)
```

Comment:
Handled in `slate/slac/kernel_builder.py`, line 86. Multiple domain support will come in a future PR.